### PR TITLE
feat(testing): add create-nodes plugin for playwright e2e targets

### PR DESCRIPTION
**Pull request by AgentEnder**
## Related Issue(s)

Fixes #

### CODE REVIEW

File: packages/playwright/src/plugins/plugin.ts [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+) {
+  const playwrightConfig: PlaywrightTestConfig = getPlaywrightConfig(
+    configFilePath,
+    context
+  );
+
+  const targetDefaults = readTargetDefaultsForTarget(
+    options.targetName,
+    context.nxJsonConfiguration.targetDefaults,
+    'executorName'
```
Comment (meeroslav) [inline packages/playwright/src/plugins/plugin.ts:69]: This should be `@nx/playwright:playwright`

File: packages/playwright/src/plugins/plugin.ts [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+          root: projectRoot,
+          projectType: 'library',
+          targets: await buildPlaywrightTargets(
+            configFilePath,
+            projectRoot,
+            normalizedOptions,
+            context
+          ),
+        },
+      },
```
Comment (mandarini) [inline packages/playwright/src/plugins/plugin.ts:57]:
```suggestion
      projects: {
        [projectRoot]: {
          root: projectRoot,
          targets: await buildPlaywrightTargets(
            configFilePath,
            projectRoot,
            normalizedOptions,
            context
          ),
        },
      },
```

we don't want the name or the type, iirc!

File: packages/playwright/src/plugins/plugin.ts [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+    const siblingFiles = readdirSync(projectRoot);
+    if (
+      !siblingFiles.includes('package.json') &&
+      !siblingFiles.includes('project.json')
+    ) {
+      return {};
+    }
+
+    const normalizedOptions = normalizeOptions(options);
+    const projectName = basename(projectRoot);
```
Comment (mandarini) [inline packages/playwright/src/plugins/plugin.ts:43]:
```suggestion

```
this can be removed

File: packages/playwright/src/plugins/plugin.ts [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+) {
+  const playwrightConfig: PlaywrightTestConfig = await loadPlaywrightConfig(
+    join(context.workspaceRoot, configFilePath)
+  );
+
+  const targetDefaults = readTargetDefaultsForTarget(
+    options.targetName,
+    context.nxJsonConfiguration.targetDefaults,
+    'nx:run-commands'
+  );
```
Comment (mandarini) [inline packages/playwright/src/plugins/plugin.ts:76]:
```suggestion

```

i think we don't want to read targetDefaults.

File: packages/playwright/src/plugins/plugin.ts [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+    cache: targetDefaults?.cache ?? true,
+    inputs:
+      targetDefaults?.inputs ?? 'production' in namedInputs
+        ? ['default', '^production']
+        : ['default', '^default'],
+    outputs:
+      targetDefaults?.outputs ?? getOutputs(projectRoot, playwrightConfig),
+    options: {
+      ...baseTargetConfig.options,
+      ...targetDefaults?.options,
```
Comment (mandarini) [inline packages/playwright/src/plugins/plugin.ts:100]: I think we don't want targetDefaults here

File: packages/playwright/src/plugins/plugin.ts [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+    },
+  };
+
+  if (options.ciTargetName) {
+    const ciTargetDefaults = readTargetDefaultsForTarget(
+      options.ciTargetName,
+      context.nxJsonConfiguration.targetDefaults,
+      'nx:run-commands'
+    );
```
Comment (mandarini) [inline packages/playwright/src/plugins/plugin.ts:109]: Same here, I think we don't want targetDefaults

File: packages/playwright/src/plugins/plugin.ts [hunk 7]
```diff
@@ -0,0 +1,254 @@
...
+    outputs.push(getOutput('./test-results'));
+  }
+
+  return outputs;
+}
+
+function normalizeOptions(options: PlaywrightPluginOptions): NormalizedOptions {
+  return {
+    ...options,
+    targetName: options.targetName ?? 'e2e',
```
Comment (mandarini) [inline packages/playwright/src/plugins/plugin.ts:252]: should there be also a default for `ciTargetName` here?

File: packages/playwright/src/generators/init/init.ts [hunk 8]
```diff
@@ -70,4 +76,25 @@ export async function initGenerator(tree: Tree, options: InitGeneratorSchema) {
...
+    !nxJson.plugins.some((p) =>
+      typeof p === 'string'
+        ? p === '@nx/playwright/plugin'
+        : p.plugin === '@nx/playwright/plugin'
+    )
+  ) {
+    nxJson.plugins.push({
+      plugin: '@nx/playwright/plugin',
+      options: {
+        targetName: 'e2e',
```
Comment (mandarini) [inline packages/playwright/src/generators/init/init.ts:93]: do we need the `ciTargetName` here as well?

File: packages/playwright/src/generators/init/init.spec.ts [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+          skipPackageJson: false,
+        });
+      }
+    );
+    const nxJson = readNxJson(tree);
+    expect(nxJson.plugins).toMatchInlineSnapshot(`
+      [
+        {
+          "options": {
+            "targetName": "e2e",
```
Comment (mandarini) [inline packages/playwright/src/generators/init/init.spec.ts:31]: do we need the `ciTargetName` here as well?

**Review by mandarini — CHANGES_REQUESTED** [review]
Left some comments!!

File: packages/playwright/src/generators/init/init.ts (hunk 8 above)
Comment (AgentEnder) [reply packages/playwright/src/generators/init/init.ts:93]: Not adding it to options as its not added in they cypress generator

**Review by mandarini — APPROVED** [review]

File: packages/playwright/src/plugins/plugin.ts [hunk 10]
```diff
@@ -0,0 +1,240 @@
...
+export interface PlaywrightPluginOptions {
+  targetName?: string;
+  ciTargetName?: string;
+}
+
+interface NormalizedOptions {
+  targetName: string;
+  ciTargetName?: string;
+}
+
```
Comment (jaysoo) [inline packages/playwright/src/plugins/plugin.ts:26]: Missing caching for targets, which will add 200 ms each time the graph is calculated without daemon.

See https://github.com/nrwl/nx/blob/master/packages/next/src/plugins/plugin.ts#L28-L52

**Review by jaysoo — APPROVED** [review]

**Pull request merged**